loopy/target/pyopencl.py

```diff
@@ -370,6 +370,38 @@ def map_power(self, expr, type_context):
 # }}}
 
 
+# {{{ filter out repeated base storages in IDIS
+
+def convert_global_temp_args_to_base_storage(kernel, idis):
+    from loopy.kernel.data import (InameArg,
+                                   TemporaryVariable)
+    new_idis = []
+
+    for idi in idis:
+        if (idi.offset_for_name is not None
+                or idi.stride_for_name_and_axis is not None
+                or issubclass(idi.arg_class, InameArg)):
+            # offset and iname => no need to filter these out
+            new_idis.append(idi)
+        else:
+            name = idi.base_name or idi.name
+            var_descr = kernel.get_var_descriptor(name)
+            if isinstance(var_descr, TemporaryVariable):
+                if var_descr.base_storage is not None:
+                    assert isinstance(var_descr, TemporaryVariable)
+                    new_idis.append(idi.copy(name=var_descr.base_storage,
+                                             base_name=var_descr.base_storage))
```

Comment:
I'm not sure I love doing this at the level of the IDI. Especially setting `base_name` troubles me, because then we can't possibly trace back who this IDI belongs to. Why not just override the right part of codegen instead?